### ISSUE TITLE
page list submenu button is missing an accessible name

### ISSUE THREAD

**Issue by carolinan** [WordPress/gutenberg#36598]
### Description

When a navigation block does not have the option "Open on click" enabled, sub menu items open on hover.

Sighted keyboard users can open the submenu by tabbing past the link of the parent menu item to the button, and activate the button.

Screen reader users can navigate to the button, but the only notification is "button collapsed", which is not enough information.
The button has no accessible name:

`<button class="wp-block-navigation__submenu-icon wp-block-navigation-submenu__toggle" aria-expanded="false"><span class="wp-block-page-list__submenu-icon wp-block-navigation__submenu-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="12" viewBox="0 0 12 12" fill="none" role="img" aria-hidden="true" focusable="false"><path d="M1.50002 4L6.00002 8L10.5 4" stroke-width="1.5"></path></svg></span></button>`

### Step-by-step reproduction instructions

1. Add a navigation block with a page list. Make sure the page list has some submenu items.
2. Got to the front of the website, and with a screen reader, navigate to the menu and try toggling the submenu buttons on and off.

### Environment info

-WordPress trunk
-Current Gutenberg trunk
Tested with Twenty Twenty-two.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by tellthemachines** [comment]
This is an issue not only with Page List submenus but also with the actual Submenu block. Working on a fix now.
